### PR TITLE
Manage the FREE_BYTE_CODE_CPTR in client side

### PR DESCRIPTION
**Pull request by polaroi8d**
- Remove element from the list which has the given `cptr` key. 
- Add the source name of the last byte code _( the general program byte code )_ 
- Commented some prints for the next PR, when I'll made a DEBUG log for it

_TODO: Implement the signal sender, which can warn the engine back about the released byte code_

JerryScript-DCO-1.0-Signed-off-by: Levente Orban orbanl@inf.u-szeged.hu

### CODE REVIEW

File: jerry-debugger/jerry-client.py [hunk 1]
```diff
@@ -68,13 +70,15 @@ def parse_source(debugger, data):
     source_name = ''
     function_name = ''
 
-    function_list = []
```
Comment (zherczeg) [inline jerry-debugger/jerry-client.py:71]: JerryDebuger should be an entity, so there is no point to take this out to global level. Global variables are not recommended in general.

File: jerry-debugger/jerry-client.py [hunk 2]
```diff
@@ -136,16 +146,14 @@ def main():
                            JERRY_DEBUGGER_SOURCE_FILE_NAME,
                            JERRY_DEBUGGER_FUNCTION_NAME,
                            JERRY_DEBUGGER_PARSE_FUNCTION,
-                           JERRY_DEBUGGER_BYTE_CODE_CPTR]:
+                           JERRY_DEBUGGER_BYTE_CODE_CPTR,
+                           JERRY_DEBUGGER_FREE_BYTE_CODE_CPTR]:
```
Comment (zherczeg) [inline jerry-debugger/jerry-client.py]: Byte code free should not be handled by the parser, it should be an independent event.

File: jerry-debugger/jerry-client.py [hunk 3]
```diff
@@ -136,16 +146,14 @@ def main():
...
 
             parse_source(debugger, data)
 
-        elif buffer_type == JERRY_DEBUGGER_FREE_BYTE_CODE_CPTR:
-            print("Free function")
-            print(data[2:buffer_size+2])
```
Comment (zherczeg) [inline jerry-debugger/jerry-client.py:145]: Exactly like this.

**Comment by polaroi8d** [comment]
Thanks @zherczeg for reviewing in your free time, I'm updated the PR following your reflection.

File: jerry-debugger/jerry-client.py [hunk 4]
```diff
@@ -96,14 +101,23 @@ def parse_source(debugger, data):
...
-            break;
+        else:
+            print('Parser error!')
+            return
 
         data = debugger.get_message()
 
-    print(function_list)
+    function_list_tmp[-1]['source'] = source_name # We know the last item in the list is the general byte code
+    debugger.function_list = list(function_list_tmp) # Copy the ready list to the global storage
```
Comment (polaroi8d) [inline jerry-debugger/jerry-client.py]: new_function_lisr

File: jerry-debugger/jerry-client.py [hunk 5]
```diff
@@ -67,21 +68,25 @@ def parse_source(debugger, data):
...
-    stack = [ {} ]
+    stack = [{}]
+    function_list_tmp = []
 
     while True:
         if data == None:
             return
 
+        if len(stack) == 0: # Break the while loop if there is no more data in the stack
+            print('Empty stack!')
```
Comment (polaroi8d) [inline jerry-debugger/jerry-client.py]: Míake them comments, add a log function later.

File: jerry-debugger/jerry-client.py [hunk 6]
```diff
@@ -32,11 +33,11 @@
 MAX_BUFFER_SIZE = 64  # Need to be the same as the jerry debugger MAX_BUFFER_SIZE
 HOST = "localhost"
 
-
 class JerryDebugger:
 
     def __init__(self):
         self.message_data = b''
+        self.function_list = []
```
Comment (polaroi8d) [inline jerry-debugger/jerry-client.py:40]: This should be a dictionary, where the key is the cptr.

File: jerry-main/main-unix.c [hunk 7]
```diff
@@ -687,6 +687,10 @@ main (int argc,
     ret_code = JERRY_STANDALONE_EXIT_CODE_FAIL;
   }
 
+  /* Send signal to client, for the general program byte code */
+#ifdef JERRY_DEBUGGER
+  jerry_debugger_send_type (JERRY_DEBUGGER_PARSE_FUNCTION);
```
Comment (zherczeg) [inline jerry-main/main-unix.c]: We really shouldn't need to send this low level message here. If we need, there is some bug in the code.

**Review by zherczeg — CHANGES_REQUESTED** [review]
LGTM after the main-unix change is removed.

File: jerry-main/main-unix.c (hunk 7 above)
Comment (polaroi8d) [reply jerry-main/main-unix.c]: Okey.

**Pull request merged**